frontend/src/components/Modal.tsx

```diff
@@ -12,18 +12,31 @@ function Modal({
   children,
 }: ModalProps) {
   const showClass = show ? 'flex' : 'hidden';
+  const ref: React.RefObject<HTMLDivElement> = useRef(null);
+
+  useEffect(() => {
+    const closeEsc = (e: KeyboardEvent) => {
+      if (show && (e.key === 'Escape')) {
+        handleClose();
+      }
+    };
+    window.addEventListener('keydown', closeEsc);
+
+    const closeClickOutside = (e: MouseEvent) => {
+      if (show && ref.current && !ref.current.contains(e.target as Node)) {
+        handleClose();
+      }
+    };
+    window.addEventListener('mousedown', closeClickOutside);
+  });
```

Comment:
Just stylistic but can these be given straight to the div as onKeyDown and onMouseDown?